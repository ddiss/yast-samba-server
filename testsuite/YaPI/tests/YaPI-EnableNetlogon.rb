# encoding: utf-8

module Yast
  class YaPIEnableNetlogonClient < Client
    def main
      # test for YaPI::Samba::EnableNetlogon()
      # $Id$
      # testedfiles: SambaConfig.pm

      Yast.import "SambaConfig"

      Yast.include self, "testsuite.rb"
      Yast.include self, "tests-common.rb"

      SambaConfig.Import(
        {
          "global" => {
            "_modified"        => "1",
            "workgroup"        => "Test",
            "domain master"    => "no",
            "security"         => "user",
            "preferred master" => "yes",
            "domain logons"    => "no",
            "local master"     => "no"
          }
        }
      )

      Yast.import "YaPI::Samba"

      TEST(lambda { YaPI::Samba.EnableNetlogon(true) }, [
        @r_common,
        @w_common,
        @x_common
      ], nil)

      nil
    end
  end
end

Yast::YaPIEnableNetlogonClient.new.main
