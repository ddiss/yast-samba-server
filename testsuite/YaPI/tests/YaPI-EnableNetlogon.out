Write	.etc.smb.value.global."domain logons" "no" true
Write	.etc.smb.value_type.global."domain logons" 0 true
Write	.etc.smb.value.global."domain master" "no" true
Write	.etc.smb.value_type.global."domain master" 0 true
Write	.etc.smb.value.global."local master" "no" true
Write	.etc.smb.value_type.global."local master" 0 true
Write	.etc.smb.value.global."preferred master" "yes" true
Write	.etc.smb.value_type.global."preferred master" 0 true
Write	.etc.smb.value.global.security "user" true
Write	.etc.smb.value_type.global.security 0 true
Write	.etc.smb.value.global.workgroup "Test" true
Write	.etc.smb.value_type.global.workgroup 0 true
Write	.etc.smb.section_type.global 0 true
Write	.etc.smb.section_type.netlogon 0 true
Write	.etc.smb.section_comment.netlogon "\n" true
Write	.etc.smb nil true
Return	nil
